### PR TITLE
fps upgrade fixes

### PR DESCRIPTION
**Pull request by samiwh**
Changes found while upgrading the fps starter project

### CODE REVIEW

File: workers/unity/Packages/com.improbable.gdk.core/Subscriptions/EntityGameObjectLinker.cs [hunk 1]
```diff
@@ -72,10 +72,10 @@ public EntityGameObjectLinker(World world)
             linkedGameObjects.Add(gameObject);
 
             linkedComponent.IsValid = true;
-            linkedComponent.Linker = this;
             linkedComponent.EntityId = entityId;
             linkedComponent.World = world;
             linkedComponent.Worker = workerSystem;
+            linkedComponent.Linker = this;
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.core/Subscriptions/EntityGameObjectLinker.cs]: Why? Does the ordering here actually matter?

File: workers/unity/Packages/com.improbable.gdk.core/Subscriptions/EntityGameObjectLinker.cs [hunk 2]
```diff
@@ -129,10 +134,13 @@ public void UnlinkGameObjectFromEntity(EntityId entityId, GameObject gameObject)
...
             var linkComponent = gameObject.GetComponent<LinkedEntityComponent>();
-            // todo check if this can happen - right now I think it can on application quit
             if (linkComponent != null)
             {
-                linkComponent.Invalidate();
+                linkComponent.IsValid = false;
+                linkComponent.EntityId = new EntityId(0);
+                linkComponent.World = null;
+                linkComponent.Worker = null;
+                linkComponent.Linker = null;
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.core/Subscriptions/EntityGameObjectLinker.cs]: This and the linking action feels like an:

`Init()` and `Deinit()` pair of methods on the linkComponent.

File: workers/unity/Packages/com.improbable.gdk.core/Subscriptions/RequiredSubscriptionsInjector.cs [hunk 3]
```diff
@@ -1,6 +1,7 @@
 using System;
 using System.Collections.Generic;
 using Improbable.Gdk.Core;
+using UnityEngine;
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.core/Subscriptions/RequiredSubscriptionsInjector.cs]: Is this actually needed?

File: workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs [hunk 4]
```diff
@@ -30,11 +30,6 @@ protected override void OnCreateManager()
 
         protected override void OnUpdate()
         {
-            if (!worker.ConnectionHandler.IsConnected())
-            {
-                return;
-            }
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs]: Why is this moved into the implementation of the `IConnectionHandler`?

File: workers/unity/Packages/com.improbable.gdk.core/Subscriptions/EntityGameObjectLinker.cs (hunk 1 above)
Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.core/Subscriptions/EntityGameObjectLinker.cs]: Side effect

Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.core/Subscriptions/EntityGameObjectLinker.cs]: The ordering does not matter

File: workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs (hunk 4 above)
Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs]: To prevent the case where you read the same diff again and again. When it says to disconnect it could also clear everything else. However you could have an implementation that clears stuff up in later frames.

File: workers/unity/Packages/com.improbable.gdk.core/Subscriptions/EntityGameObjectLinker.cs (hunk 2 above)
Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.core/Subscriptions/EntityGameObjectLinker.cs]: Discussed offline. Agreed but will merge this in for now.
